pyanaconda/anaconda.py

```diff
@@ -208,25 +201,22 @@ def intf(self):
     def initInterface(self):
         if self._intf:
             raise RuntimeError("Second attempt to initialize the InstallInterface")
-
-        if self.webui_mode:
```

Comment:
I don't think it is necessary to move the code. I would just replace the condition with:
`if self.gui_mode and os.path.exists("/usr/share/cockpit/anaconda-webui"):`

This can change again in the future and it is more readable that way. It will also fix the issue with `addon_paths`, because in your version, the installer will look for add-ons written for GUI while running in the Web UI mode.

Reply:
+1, keeping it as if (webui conditions), elif (ui conditions), elif... makes most sense.

Reply:
@poncovka Here I'm not sure if such a situation can occur. Can anaconda-webui even be installed on a system without a GUI? Because if not, this condition should always ensure an error.

Reply:
> @poncovka Here I'm not sure if such a situation can occur. Can anaconda-webui even be installed on a system without a GUI? Because if not, this condition should always ensure an error.

It has to be. The shared code and the back-end are part of the `anaconda-core` package. The `anaconda-gui` and `anaconda-tui` packages contain only code related to GUI and TUI. It is definitely possible to have a boot.iso with TUI - GUI. It is not posible to have a boot.iso with GUI - TUI due to the VNC support and the rescue mode. Anyway, having a boot.iso with Web UI + TUI - GUI will be inevitable in the near future.